### PR TITLE
Add leave team button

### PR DESCRIPTION
**Pull request by nanaya**
Apart of the message that needs fixing, I wonder if the route should have team id in it... (there's also question of if it's worth adding an OsuAuthorize check for this)

### CODE REVIEW

File: resources/views/teams/show.blade.php [hunk 1]
```diff
@@ -13,6 +13,12 @@
     $teamMembers['member'] ??= [];
     $teamMembers['leader'] ??= $toJson([$team->members()->make(['user_id' => $team->leader_id])->userOrDeleted()]);
     $headerUrl = $team->header()->url();
+
+    $currentUser = Auth::user();
+    $buttons = new Ds\Set();
+    if ($currentUser !== null && $currentUser->team?->getKey() === $team->getKey() && $currentUser->getKey() !== $team->leader_id) {
+        $buttons->add('part');
+    }
```
Comment (notbakaneko) [inline resources/views/teams/show.blade.php:21]: Are there more buttons coming 👀

File: app/Http/Controllers/TeamsController.php [hunk 2]
```diff
@@ -21,6 +28,22 @@ public function edit(string $id): Response
...
 
+    public function part(): Response
+    {
+        $member = TeamMember::findOrFail(\Auth::user()->getKey());
+
+        if ($member === null) {
+            \Session::flash('popup', osu_trans('teams.part.no_team'));
+
+            return ujs_redirect(route('home'));
+        }
```
Comment (notbakaneko) [inline app/Http/Controllers/TeamsController.php:39]: always going to be false with `findOrFail` above

**Review by notbakaneko — CHANGES_REQUESTED** [review]

File: resources/views/teams/show.blade.php (hunk 1 above)
Comment (nanaya) [reply resources/views/teams/show.blade.php:21]: why yes 👀

File: app/Http/Controllers/TeamsController.php (hunk 2 above)
Comment (notbakaneko) [reply app/Http/Controllers/TeamsController.php]: localization key also unused

File: resources/views/teams/show.blade.php [hunk 3]
```diff
@@ -69,6 +75,21 @@ class="btn-circle btn-circle--page-toggle"
...
+        @if (!$buttons->isEmpty())
+            <div class="profile-detail-bar profile-detail-bar--team">
+                @if ($buttons->contains('part'))
+                    <form
+                        action="{{ route('teams.part') }}"
+                        data-turbo-confirm="{{ osu_trans('common.confirmation') }}"
+                        method="POST"
+                    >
+                        <button class="team-action-button team-action-button--part">
+                                {{ osu_trans('teams.show.bar.part') }}
```
Comment (notbakaneko) [inline resources/views/teams/show.blade.php:87]: too many idents btw

**Comment by nanaya** [comment]
sec, I think it's more convenient to just add an OsuAuthorize check

**Review by notbakaneko — APPROVED** [review]

**Pull request merged**